### ISSUE TITLE
[IOperation] Should As New initializers be the same node?

### ISSUE THREAD

**Issue by 333fred** [dotnet/roslyn#18335]
Consider the following VB code:

```VB.NET
Module Program
    Sub Main(args As String())
        Dim i1, i2 As New Integer
    End Sub
End Module
```

Should the initializers for the `IVariableDeclaration`s be literally the same instance, or should there be some way to tell if the initializer for two different variable declarations is the same initializer? If they are the same object, it will complicate the creation of a parent pointer.

**Comment by jinujoseph** [comment]
**Design Meeting Notes** 

From @CyrusNajmabadi 

The idea behind the IOperation API for variables is that it will more faithfully represent how VB treats variables, while avoiding problems around parent pointers and the like.  
So, If you have:

Dim x, y as New Z(), m, n as New W()

Then that would be a ’IVariableDeclarationStatement’ with an array of two ‘IDeclarationGroup’s.  The two groups would represent “x, y as New Z()” and “m, n as New W()” respectively.

Each IDeclarationGroup would have two ‘IVariableDeclaration’s *and* an optional Initializer.  In this case, the Initializer for the first group would be “new Z()” and the initializer for the second group would be “New W()”.

Each IVariableDeclaration then has an optional initializer as well.  In this example, they would all be null.  However, if you had: “Dim x = 1, y = 2”, then the Initializer for the IVariableDeclaration for ‘x’ would be ‘1’ and the Initializer for ‘y’ would be ‘2’.

--

Specifically, in the first example “new Z()” is not reachable from the IVariableDeclaratoin for ‘x’ and ‘y’.  It is only reachable for the IDeclaratoinGroup that contains ‘x’ and ‘y’.

--

For C# most of this API would remain null.  i.e. C# can only really do :   “var i = 0, j = 1”.   So it would only ever have a single declaration group.  That declaration group would never have an initializer.  And each variable declaration would have an initializer here.

**Comment by CyrusNajmabadi** [comment]
Note: none of the names are set in stone.  The above is just about defining the general shape we'd like.  If you have ideas about good names here, def let us know.

**Comment by jinujoseph** [comment]
Fixed via #18698
